lychee-lib/src/filter/mod.rs

```diff
@@ -22,63 +21,65 @@ impl Filter {
         excludes: Option<Excludes>,
         scheme: Option<String>,
     ) -> Self {
-        let includes = match includes {
-            Some(includes) => includes,
-            None => Includes::default(),
-        };
-        let excludes = match excludes {
-            Some(excludes) => excludes,
-            None => Excludes::default(),
-        };
         Filter {
-            includes,
-            excludes,
+            includes: includes.unwrap_or_default(),
+            excludes: excludes.unwrap_or_default(),
             scheme,
         }
     }
 
-    pub fn excluded(&self, request: &Request) -> bool {
+    pub fn is_excluded(&self, uri: &Uri) -> bool {
         // Skip mail?
-        if self.excludes.is_mail_excluded() && matches!(request.uri, Uri::Mail(_)) {
+        if self.excludes.is_mail_excluded() && matches!(uri, Uri::Mail(_)) {
             return true;
         }
         // Skip specific IP address?
-        if self.excludes.ip(&request.uri) {
+        if self.excludes.ip(&uri) {
             return true;
         }
-        // No regex includes/excludes at all?
-        if self.includes.is_empty() && self.excludes.is_empty() {
-            // Not excluded unless it's a known false positive
-            return self.excludes.false_positive(request.uri.as_str());
-        }
-        // Includes take precedence over excludes
-        if self.includes.regex(request.uri.as_str()) {
+
+        let input = uri.as_str();
+        if self.includes.is_empty() {
```

Comment:
Here is one thing you haven't mentioned.

From L42 to L66 is some non-obvious optimization.
There were duplicated checks because some previous checks implied some latter ones.

The conditionals here are carefully organized so it doesn't miss any case while it won't check the same thing twice.

Reply:
Yeah, I saw that but I forgot to mention it. Great work!